packages/base/catalog-entry.gts

```diff
@@ -52,12 +53,13 @@ export class CatalogEntry extends Card {
   @field description = contains(StringCard);
   @field ref = contains(CardRefCard);
   @field isPrimitive = contains(BooleanCard, { computeVia: async function(this: CatalogEntry) {
-    let module: Record<string, any> = await import(this.ref.module);
-    let Clazz: typeof Card = module[this.ref.name];
-    return primitive in Clazz ||
-      // the base card is a special case where it is technically not a primitive, but because it has no fields
-      // it is not useful to treat as a composite card (for the purposes of creating new card instances).
-      (baseCardRef.module === this.ref.module && baseCardRef.name === this.ref.name);
+    let card: typeof Card | undefined = await loadCard(this.ref);
```

Comment:
hmmmm, the idea here is that we want to have the card authors be able to use `await import(blah)`, but that it is compiled to `Loader.import(blah)`. I think instead of making this change, you need to update the babel transpilation so that instead of compiling in `Loader.import(blah)` we compile in loadCard(blah), such that card authors do not need to know about our special API around importing cards and can just use the normal dynamic import to load a card.

Reply:
Not exactly though. Transpiling `import()` to `Loader.import()` is safe because they mean exactly the same thing. `loadCard` doesn't mean the same thing. It's a layer higher.

Even if we do transpire the thing that card authors write, they should write something that's clearly asking to load a card from a cardref, not a module from a URL.